torch/_inductor/async_compile.py

```diff
@@ -213,10 +213,12 @@ def cpp_pybinding(self, argtypes: List[str], source_code: str):
             )
             return LambdaFuture(get_result)
 
-    def cuda(self, source_code, dst_file_ext):
+    def cuda(self, source_code, dst_file_ext, aot_compile=False):
         kernel_code_log.info("CUDA Kernel:\n%s", source_code)
 
         def task():
+            if aot_compile:
+                output_path, *_ = CUDACodeCache.compile(source_code, "o")
```

Comment:
I would assume to ensure that a .o file is being built and inserted into the cache, even if dst_file_ext is, let's say "exe" or "so". So, if dst_file_ext is different from "o", it will mean that nvcc has to compile it all twice, which increases compilation time. Is this intentional?

Reply:
Yes, the doubled compilation time is intentional atm. One to compile a ".so" and another to compile a ".o". We need the ".so" for running Inductor and the ".o" for AOTInductor. I guess it wouldn't be too too bad as long as they're compiled on separate threads.